lib/programmemory.h

```diff
@@ -110,21 +110,19 @@ struct CPPCHECKLIB ProgramMemory {
 
     void setValue(const Token* expr, const ValueFlow::Value& value);
     const ValueFlow::Value* getValue(nonneg int exprid, bool impossible = false) const;
+    ValueFlow::Value* getValue(nonneg int exprid, bool impossible = false);
 
-    bool getIntValue(nonneg int exprid, MathLib::bigint& result) const;
+    bool getIntValue(nonneg int exprid, MathLib::bigint& result, bool impossible = false) const;
     void setIntValue(const Token* expr, MathLib::bigint value, bool impossible = false);
 
-    bool getContainerSizeValue(nonneg int exprid, MathLib::bigint& result) const;
-    bool getContainerEmptyValue(nonneg int exprid, MathLib::bigint& result) const;
+    bool getContainerSizeValue(nonneg int exprid, MathLib::bigint& result, bool impossible = false) const;
+    bool getContainerEmptyValue(nonneg int exprid, MathLib::bigint& result, bool impossible = false) const;
```

Comment:
I flipped the default behavior on this but addressed it via an explicit parameter in its only usage in `vf_analyzers.cpp`.

Reply:
Dont add the `impossible` parameters. They dont make sense at all.